### PR TITLE
#2892 - TestKit: create child actors via TestProbe

### LINKED ISSUE #2892: Port TestKit parent actor utilities
See: https://github.com/akka/akka/pull/21201.

We are currently missing these TestKit facilities, And they should be ported.

### PR DESCRIPTION
**Pull request by nagytech**
Closes #2892

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
@nagytech needs API approval: https://getakka.net/community/public-api-changes.html

File: docs/articles/actors/testing-actor-systems.md [hunk 1]
```diff
@@ -179,6 +179,11 @@ The first option is to avoid use of the `context.parent` function and create a c
 
 [!code-csharp[DependentChild](../../../src/core/Akka.Docs.Tests/Testkit/ParentSampleTest.cs?range=44-57)]
 
+###Create the child using the TestProbe
+The `TestProbe` class can directly create child actors using the `ChildActorOf` methods.  
+
+[!code-csharp[TestProbeChild](../../../src/core/Akka.Docs.Tests/Testkit/ParentSampleTest.cs?range=60-64)]
```
Comment (Aaronontheweb) [inline docs/articles/actors/testing-actor-systems.md]: I'd change this to use `#region` targeting instead, i.e. https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka.Docs.Tests/Streams/StreamRefsDocTests.cs#L20 - you can actually target the region name using DocFx: https://dotnet.github.io/docfx/spec/docfx_flavored_markdown.html#code-snippet-sample

File: src/core/Akka.TestKit/TestActor.cs [hunk 2]
```diff
@@ -118,5 +119,53 @@ public class SetAutoPilot : INoSerializationVerificationNeeded
...
             public AutoPilot AutoPilot { get { return _autoPilot; } }
         }
+
+        /// <summary>
+        /// Message which is intended to allow TestKit to spawn a child actor
+        /// </summary>
+        public class Spawn : INoSerializationVerificationNeeded
+        {
+            /// <summary>
+            /// TBD
```
Comment (Aaronontheweb) [inline src/core/Akka.TestKit/TestActor.cs]: I'd just leave off the `TBDs` rather than add new ones.

File: src/core/Akka.TestKit/TestKitBase.cs [hunk 3]
```diff
@@ -450,6 +452,54 @@ protected virtual void Shutdown(ActorSystem system, TimeSpan? duration = null, b
...
         }
 
+        /// <summary>
+        /// Spawns an actor as a child of this test actor, and returns the child's IActorRef
+        /// </summary>
+        /// <param name="props">Child actor props</param>
+        /// <param name="name">Child actor name</param>
+        /// <param name="supervisorStrategy">Supervisor strategy for the child actor</param>
+        /// <returns></returns>
+        public IActorRef ChildActorOf(Props props, string name, SupervisorStrategy supervisorStrategy)
```
Comment (Aaronontheweb) [inline src/core/Akka.TestKit/TestKitBase.cs]: These all look good to me

File: docs/articles/actors/testing-actor-systems.md (hunk 1 above)
Comment (Aaronontheweb) [reply docs/articles/actors/testing-actor-systems.md]: Reason why I recommend it: re-organizing the file doesn't break documentation as easily.

File: docs/articles/actors/testing-actor-systems.md [hunk 4]
```diff
@@ -187,7 +192,7 @@ If you prefer to avoid modifying the parent or child constructor you can create
 ###Externalize child making from the parent
 Alternatively, you can tell the parent how to create its child. There are two ways to do this: by giving it a `Props` object or by giving it a function which takes care of creating the child actor:
 
-[!code-csharp[FabrikatedParent](../../../src/core/Akka.Docs.Tests/Testkit/ParentSampleTest.cs?range=84-103)]
+[!code-csharp[FabrikatedParent](../../../src/core/Akka.Docs.Tests/Testkit/ParentSampleTest.cs?range=94-110)]
```
Comment (Aaronontheweb) [inline docs/articles/actors/testing-actor-systems.md]: Same feedback as above.

File: docs/articles/actors/testing-actor-systems.md (hunk 1 above)
Comment (nagytech) [reply docs/articles/actors/testing-actor-systems.md]: The latest changes should be using DocFX named references now.  

To help convert the remaining docs, I've written a program to go through all the .md files and automatically convert line references to C# regions.  I think there's a few references that use `#` syntax, so might need to be adapted in the future. 

https://gist.github.com/nagytech/bb07f1c46cda7b88ed8c3a5f4df7e00b

**Comment by Aaronontheweb** [comment]
@nagytech still need to commit your API approvals - last thing we need before I can merge this in.

**Comment by nagytech** [comment]
@Aaronontheweb I just noticed this was pending for '1 change requested'.  But I can't see any outstanding comments that I would need to resolve.

**Comment by Aaronontheweb** [comment]
@nagytech I thought it was because the API approvals were missing, but maybe I'm mistaken about that

**Pull request merged**